google/auth/action.yml

```diff
@@ -0,0 +1,21 @@
+name: google/auth
+
+description: |
+  This is an opinionated GitHub Action to authenticate with GCP.
+  It generates a Workload Identity Pool Provider ID based on the repository name, which is compatible with the
+  GCP Workload Identity Pool Provider ID we use for Elastic Observability repositories.
+
+runs:
+  using: composite
+  steps:
+    - name: Generate workloadIdentityPool provider ID
+      id: generate-workload-identity-pool-provider-id
+      run: |
+        HASH=$(echo -n "${GITHUB_REPOSITORY}" | sha256sum | cut -c1-27)
+        echo "workload_identity_provider_id=repo-${HASH}" >> "${GITHUB_OUTPUT}"
```

Comment:
A few questions:

- Do we want to support Windows? 
- Should the output be exported too?

Reply:
> Do we want to support Windows?

Do we have an existing use case for Windows? If not, I'd rather add it when it becomes necessary.
Otherwise, we slightly increase the complexity for no reason. 

> Should the output be exported too?

I don't see why one would need the workload identity pool id.
I might be wrong, but I think we should only add it if we already see a reason for it. Otherwise, we can always add it later.


Of course, if you already know any of this is needed, I'm happy to add it.

Reply:
With the latest changes, I actually addressed both points.

- Added outputs to be able to test the output
- Changed the hash-generating script to Python to make it cross-platform compatible.